core/frontend/src/render/webgl/Texture.ts

```diff
@@ -555,13 +606,16 @@ export class ExternalTextureLoader { /* currently exported for tests only */
 
     try {
       if (!req.imodel.isClosed) {
-        const maxTextureSize = System.instance.capabilities.maxTextureSize;
-        const texBytes = await req.imodel.getTextureImage({ name: req.name, maxTextureSize });
-        if (undefined !== texBytes) {
-          const imageSource = new ImageSource(texBytes, req.format);
-          const image = await imageElementFromImageSource(imageSource);
+        const maxTextureSize = System.instance.capabilities.maxTexSizeAllow;
+        const texData = await req.imodel.queryTextureImage({ name: req.name, maxTextureSize });
+        if (undefined !== texData) {
+          const cnvReq = { req, texData };
+          this._convertRequests.push(cnvReq);
+          do {
+            if (!this._convertPending)
+              await this._convertTexture();
+          } while (!this._convertPending && this._convertRequests.length > 0);
           if (!req.imodel.isClosed) {

```

Comment:
@pmconne Is this test necessary here?

Reply:
Yes. After you do something async, you should check that the iModel wasn't meanwhile closed (or more generally, that the state you had before you `await`ed remains valid).

Reply:
I get that concept, but at this point in the code it is not dependent on anything from the imodel anymore.  Which made me wonder if this particular test was necessary?

Reply:
WebGL textures produced from queryTextureImage are cached on the iModel and freed when it is closed. If the texture was deleted, there's no point in proceeding further here.

Reply:
Ah okay, thanks.